app/src/ereferrals/java/org/eyeseetea/malariacare/data/remote/PullOrganisationCredentials.java

```diff
@@ -0,0 +1,127 @@
+package org.eyeseetea.malariacare.data.remote;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+import org.eyeseetea.malariacare.data.database.model.Program;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.exception.InvalidCredentialsException;
+import org.eyeseetea.malariacare.domain.exception.NetworkException;
+import org.eyeseetea.malariacare.domain.exception.PullConversionException;
+import org.eyeseetea.malariacare.network.ServerAPIController;
+import org.hisp.dhis.client.sdk.models.attribute.AttributeValue;
+import org.hisp.dhis.client.sdk.models.organisationunit.OrganisationUnit;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class PullOrganisationCredentials {
```

Comment:
@manuelplazaspalacio This intermediate class is necessary? I think that ServerAPIController should return dhis objects directly not json.